skippkg-finder.py

```diff
@@ -325,21 +341,26 @@ def crawl(self):
                         binary not in self.skiplist_supplement_ignores:
                     obsoleted.append(binary)
 
-        skip_list = ET.Element('group', {'name': 'NON_FTP_PACKAGES'})
-        ET.SubElement(skip_list, 'conditional', {'name': 'drop_from_ftp'})
-        packagelist = ET.SubElement(skip_list, 'packagelist', {'relationship': 'requires'})
-        for pkg in sorted(obsoleted):
-            if not self.print_only and self.verbose:
-                print(pkg)
-            attr = {'name': pkg}
-            ET.SubElement(packagelist, 'package', attr)
+        skip_list = self.create_group('NON_FTP_PACKAGES', 'drop_from_ftp', obsoleted)
+
+        # Handle the conditionals
+        cond_list = {}
+        for item in self.skiplist_conditionals:
+            # node[0] is the condition, node[1] is the package
+            # an example of the format: only_x86_64:glibc-32bit
+            node = item.split(':')
+            if node[0] not in cond_list:
+                cond_list[node[0]] = []
+            cond_list[node[0]].append(node[1])
+        for cond in cond_list.keys():
+            group = self.create_group('NON_FTP_PACKAGES_' + cond, cond, cond_list[cond])
+            skip_list += group
```

Comment:
You're appending multiple XML documents which is technically not valid. Can you do the appending on the ET level?

Reply:
FWICT you end up with a document like

```
<group>
   stuff...
</group>
<group>
   more stuff...
</group>
```

That is not a valid XML document and can't be used with `<xi:include/>`. If it works, I suppose OBS ignores the spec or implements it wrongly.

Reply:
yes, it's not a valid xml format, but it works with product-builder, multiple groups in a file is allowed, like `dvd.group` hjas multiple group tree as `dvd.x86_64`, `dvd.s390x`, etc,.

Reply:
Sigh. Even with `<?xml ...` in between?

Reply:
> Can you do the appending on the ET level?

Can't do it in the one file(NON_FTP_PACKAGES.group), a `group` tree can only have an `conditional` element, in case want to make every group file to be an standard xml format, then we need multiple NON_FTP_PACKAGES_$contidions.group file for each `group`.

Reply:
*shrug* this is all chaotic anyway